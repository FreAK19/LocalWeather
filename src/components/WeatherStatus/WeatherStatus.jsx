import React from 'react';
import PropTypes from 'prop-types';
import './WeatherStatus.less'

const WeatherStatus = ({icon}) => {
  const className = `pe-is-w-${icon} pe-4x pe-va`;
  return (
    <div className='weather__status'>
      <h3>Weather / Weather Status</h3>
      <p>
        <i className={className} />
      </p>
    </div>
)
};

WeatherStatus.propTypes = {icon: PropTypes.string.isRequired};

export default WeatherStatus;
