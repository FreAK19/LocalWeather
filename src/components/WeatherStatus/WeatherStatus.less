@viewport-size-tablet: 640px;

.weather__status {
  flex-basis: 40%;

  @media screen and (max-width: @viewport-size-tablet) {
    & {
      flex-basis: initial;
      margin: auto;
    }
  }

  & > h3 {
    margin-top: 0;
    font-weight: 400;
    text-transform: capitalize;
  }
}

.weather__icon {
  font-size: 6rem;
}