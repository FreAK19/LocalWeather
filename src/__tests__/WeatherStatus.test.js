import React from 'react';
import WeatherStatus from '../components/WeatherStatus/WeatherStatus';

describe('WeatherStatus', () => {
  test('WeatherStatus should render correctly', () => {
    const component = shallow(<WeatherStatus icon="pe-is-w-rain" status="Rain" />);
    expect(component).toMatchSnapshot();
  });

  test('WeatherStatus should contain icon element with specific className', () => {
    const component = shallow(<WeatherStatus icon="pe-is-w-rain" status="Rain" />);
    const expected = component.find('i').hasClass('pe-is-w-rain weather__icon');
    expect(expected).toBe(true);
  });

  test('WeatherStatus will take correct status', () => {
    const component = shallow(<WeatherStatus icon="pe-is-w-rain-and-snow" status="Heavy rain and snow" />);
    const actual = component.find('h3').text();
    const expected = 'Heavy rain and snow';
    expect(actual).toMatch(expected);
  });
});
