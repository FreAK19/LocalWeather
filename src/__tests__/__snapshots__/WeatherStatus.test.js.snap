// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherStatus WeatherStatus should render correctly 1`] = `
<div
  className="weather__status"
>
  <h3>
    Rain
  </h3>
  <p>
    <i
      className="pe-is-w-rain weather__icon"
    />
  </p>
</div>
`;

exports[`WeatherStatus should render correctly 1`] = `
<div
  className="weather__status"
>
  <h3>
    Rain
  </h3>
  <p>
    <i
      className="pe-is-w-rain weather__icon"
    />
  </p>
</div>
`;
